stage #2/module #3 Collections/lesson #1. Overview/overview.md

```diff
@@ -0,0 +1,332 @@
+specific data structure. Programmer can simply use the best implementation to drastically boost the performance
+of algorithm/program.
+
+* **Consistent API.**
+
+* **Reduces effort to design new APIs.** You don't have to reinvent the wheel each time are
+creating an API that relies on collections; instead, just use standard collection
+interfaces.
+
+* **Fosters software reuse.** Collections interfaces and implementations are flexible
+enough to be reused given that there is a c ontract of interface.
+
+##Revision
+* **Class** - A class is a user
+  defined blueprint or prototype from which objects are created. It represents
+  the set of properties or methods that are common to all objects of one type.
+  Like
+* **Interface** - Like a class, an interface can have methods and variables, but the methods declared in an
+  interface are by default abstract ( only method signature, no body ). Interfaces specify what a
+  class must do and not how. It is the blueprint of the class.
+```
+public interface MyInterface {
+    //don’t need to implement, since java 8
+    default void newMethod() {
+        System.out.println("Newly added default method");
+    }
+
+    //don’t need to implement + we cannot override it
+    static void anotherNewMethod() {
+        System.out.println("Newly added static");
+    }
+
+    //must be implemented
+    void existingMethod(String str);
+}
+```
+
+##Hierarchy
+![](./media/hierarchy.png)
+
+##Iterable interface
+The **Iterable** interface was introduced in JDK 1.5. It belongs to java.lang package. In general, an object Implementing Iterable allows it to be iterated. An iterable interface allows an object to be the target of enhanced for loop(**for-each loop**).
+
+**Iterable** interface is the root interface for all the collection classes
+
+```
+public interface Iterable<T>{
+  Iterator<T>    iterator();
+  
+  Spliterator<T> spliterator();
+
+  void forEach(Consumer<? super T> action);
+}
+
+```
+Method | Second Description
+------------ | -------------
+**Iterator T > iterator()** | Returns the iterator
+
+There are three ways in which objects of Iterable can be iterated:
+* Using enhanced for loop(for-each loop)
+``` 
+List<String> list = new ArrayList><();
+
+list.add("one");
+list.add("two");
+list.add("three");
+
+for( String element : list ){
+    System.out.println( element.toString() );
+}
+```
+* Using Iterable forEach loop
+```
+List<String> list = new ArrayList><();
+
+list.add("one");
+list.add("two");
+list.add("three");
+
+list.forEach( (element) -> {
+    System.out.println( element );
+});
+```
+* Using Iterator<T> interface
+```
+List<String> list = new ArrayList><();
+
+list.add("one");
+list.add("two");
+list.add("three");
+
+Iterator<String> iterator = list.iterator();
+
+while(iterator.hasNext()) {
+    String element = iterator.next();
+    System.out.println(element);
+}
+```
+##Iterator
```

Comment:
It seems that a space is needed here to format the title (## Iterator)
And in other places too